File: python/anagram/anagram.py
def is_anagram(word, candidate):
  word = word.lower()
  candidate = candidate.lower()

  return word != candidate and sorted(word) == sorted(candidate)

def detect_anagrams(word, candidates):
  anagrams = []

  for candidate in candidates:
    if is_anagram(word, candidate):
      anagrams.append(candidate)

  return anagrams

File: go/anagram/anagram.go
package anagram

import (
	"sort"
	"strings"
)

// SortedString returns are string with all of the characters sorted
func SortedString(s string) string {
	s = strings.ToLower(s)
	chars := strings.Split(s, "")
	sort.Strings(chars)
	return strings.Join(chars, "")
}

// Detect possible anagrams given a word and a list of candidates
func Detect(subject string, candidates []string) (anagrams []string) {
	if len(subject) == 0 {
		return
	}

	for _, candidate := range candidates {
		if !strings.EqualFold(subject, candidate) {
			if SortedString(candidate) == SortedString(subject) {
				anagrams = append(anagrams, candidate)
			}
		}
	}
	return
}
